### PR TITLE
Restrict the non joinable beacons

### PR DESCRIPTION
**Pull request by dgowri**
Respond to the beacon requests with the beacon only if they are joinable to
reduce the traffic.

### CODE REVIEW

File: src/core/mac/mac.cpp [hunk 1]
```diff
@@ -1655,5 +1668,20 @@ void Mac::ResetCounters(void)
...
+bool Mac::IsBeaconJoinable(void)
+{
+    uint8_t numUnsecurePorts;
+
+    if (mNetif.GetKeyManager().GetSecurityPolicyFlags() & OT_SECURITY_POLICY_BEACONS)
+    {
+        // set the Joining Permitted flag
+        mNetif.GetIp6Filter().GetUnsecurePorts(numUnsecurePorts);
+
+        return (numUnsecurePorts ? true : false);
```
Comment (abtink) [inline src/core/mac/mac.cpp]: We should have a single `return`  at end of function (per OT coding style)

File: src/core/mac/mac.hpp [hunk 2]
```diff
@@ -592,6 +592,12 @@ class Mac
     void ResetCounters(void);
 
     /**
+     * This method will tell us if the beacon resp has joinable set or not
+     *
+     */
```
Comment (abtink) [inline src/core/mac/mac.hpp]: nit: I suggest adding either doxygen comment @returns, or @retval to document return value of a non-void method/function.

File: src/core/openthread-core-default-config.h [hunk 3]
```diff
@@ -726,6 +726,17 @@
...
 
 /**
+ * @def OPENTHREAD_CONFIG_ENABLE_BEACON_RSP_IF_JOINABLE
+ *
+ * Define to 1 if you want to enable beacon response for joinable networks only,
+ * by default its value is 0
+ *
+ */
+#ifndef OPENTHREAD_CONFIG_ENABLE_BEACON_RSP_IF_JOINABLE
+#define OPENTHREAD_CONFIG_ENABLE_BEACON_RSP_IF_JOINABLE             0
```
Comment (abtink) [inline src/core/openthread-core-default-config.h]: nit: style to align the `0` with other macro definitions.

File: src/core/mac/mac.cpp [hunk 4]
```diff
@@ -1655,5 +1668,20 @@ void Mac::ResetCounters(void)
...
+    }
```
Comment (dgowri) [inline src/core/mac/mac.cpp]: Agree, Fixed

File: src/core/mac/mac.cpp (hunk 1 above)
Comment (dgowri) [inline src/core/mac/mac.cpp]: Fixed

File: src/core/mac/mac.hpp (hunk 2 above)
Comment (dgowri) [inline src/core/mac/mac.hpp]: Fixed

File: src/core/openthread-core-default-config.h (hunk 3 above)
Comment (dgowri) [inline src/core/openthread-core-default-config.h]: Fixed

File: src/core/mac/mac.cpp [hunk 5]
```diff
@@ -1574,6 +1574,19 @@ ThreadError Mac::HandleMacCommand(Frame &aFrame)
         mCounters.mRxBeaconRequest++;
         otLogInfoMac(GetInstance(), "Received Beacon Request");
 
+#if OPENTHREAD_CONFIG_ENABLE_BEACON_RSP_IF_JOINABLE
+
+        if (!IsBeaconJoinable())
+        {
+            mBeaconsEnabled = false;
```
Comment (dgowri) [inline src/core/mac/mac.cpp]: Fixed

**Comment by dgowri** [comment]
I signed it.

**Pull request merged**